### PR TITLE
feat(catalog): introduce user id for user info

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

As title, introduce user id for user info and use id to communicate with meta. Besides I also changed catalog owner to id. This is a pre-work of #3998  to support like `ALTER USER name RENAME TO new_name`.

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/frontend/src/user/user_manager.rs [hunk 1]
```diff
@@ -16,45 +16,59 @@ use std::collections::HashMap;
...
 
-use crate::user::{UserInfoVersion, UserName};
+use crate::user::{UserId, UserInfoVersion};
 
 /// `UserInfoManager` is responsible for managing users.
 pub struct UserInfoManager {
     version: UserInfoVersion,
-    users: HashMap<UserName, UserInfo>,
+    user_by_name: HashMap<String, UserInfo>,
+    user_name_by_id: HashMap<UserId, String>,
```
Comment (xiangjinwu) [inline src/frontend/src/user/user_manager.rs:25]: Is it more natural to map name to id, and then id to info?

File: src/frontend/src/test_utils.rs [hunk 2]
```diff
@@ -446,13 +450,21 @@ impl UserInfoWriter for MockUserInfoWriter {
...
             name: DEFAULT_SUPPER_USER.to_string(),
             is_supper: true,
             can_create_db: true,
             can_login: true,
             ..Default::default()
         });
-        Self { user_info }
+        Self {
+            user_info,
+            id: AtomicU32::new(1 + RESERVED_USER_ID as u32),
```
Comment (xiangjinwu) [inline src/frontend/src/test_utils.rs:462]: I see two places using `1 + RESERVED_USER_ID` and no places using `RESERVED_USER_ID`, and a future potential caller may forget to `+1`. Is it better to define sth like `NONRESERVED_USER_ID` instead?

File: src/frontend/src/catalog/pg_catalog/pg_namespace.rs [hunk 3]
```diff
@@ -23,5 +23,5 @@ pub const PG_NAMESPACE_TABLE_NAME: &str = "pg_namespace";
 pub const PG_NAMESPACE_COLUMNS: &[PgCatalogColumnsDef] = &[
     (DataType::Int32, "oid"),
     (DataType::Varchar, "nspname"),
-    (DataType::Varchar, "nspowner"), // TODO: using user id instead and support acl.
+    (DataType::Int32, "nspowner"),
```
Comment (xiangjinwu) [inline src/frontend/src/catalog/pg_catalog/pg_namespace.rs:26]: Are we supposed to have any tests on these 2 pg_catalog things? It feels fragile that no tests need to be updated in this PR given this change.

File: src/meta/src/manager/user.rs [hunk 4]
```diff
@@ -39,38 +41,53 @@ pub struct UserManager<S: MetaStore> {
...
-    pub fn get_user_info(&self) -> &HashMap<UserName, UserInfo> {
+    fn create_user(&mut self, user: UserInfo) {
+        self.all_users.insert(user.name.clone());
+        self.user_info.insert(user.id, user);
+    }
+
+    fn drop_user(&mut self, user_id: UserId) {
+        if let Some(user) = self.user_info.remove(&user_id) {
+            self.all_users.remove(&user.name);
+        }
```
Comment (xiangjinwu) [inline src/meta/src/manager/user.rs:87]: Add some comments that user in `user_grant_relation` (as key or value) are already checked before entering this function.

File: src/meta/src/manager/user.rs [hunk 5]
```diff
@@ -553,7 +571,7 @@ mod tests {
                 .map(|&action| ActionWithGrantOption {
                     action: action as i32,
                     with_grant_option,
-                    granted_by: DEFAULT_SUPPER_USER.to_string(),
+                    granted_by: 1,
```
Comment (xiangjinwu) [inline src/meta/src/manager/user.rs:574]: Intentionally not using the const?

**Review by xiangjinwu — COMMENTED** [review]
Generally lgtm. Just some nits:

File: src/meta/src/manager/user.rs (hunk 5 above)
Comment (yezizp2012) [reply src/meta/src/manager/user.rs:574]: Yes, should use the const instead.

File: src/frontend/src/user/user_manager.rs (hunk 1 above)
Comment (yezizp2012) [reply src/frontend/src/user/user_manager.rs:25]: Well, I followed the mapping rule as same as all `Catalog`'s definition. We can fix all this together in a new PR.

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**